File: my-folder/0121-best-time-to-buy-and-sell-stock/solution.c
int maxProfit(int* prices, int pricesSize) {
    int min_price = prices[0];
    int max_profit = 0;
    for (size_t i = 1; i < pricesSize; i++){
        if (min_price > prices[i])
            min_price = prices[i];
        else {
            int profit = (prices[i] - min_price);
            if (max_profit < profit)
                max_profit = profit;
        }
    }
    return max_profit;
}

File: my-folder/0121-best-time-to-buy-and-sell-stock/solution.rs
impl Solution {
    pub fn max_profit(prices: Vec<i32>) -> i32 {
        use std::cmp::max;
        use std::cmp::min;
        let mut min_price = prices[0];
        let mut max_profit = 0;
        for price in prices.into_iter().skip(1) {
            min_price = min(min_price, price);
            max_profit = max(max_profit, price - min_price);
        }
        max_profit
    }
}
